### PR TITLE
ReflectAll allow to accept object of functions

### PR DESCRIPTION
**Pull request by andalm**
I made this PR for solve this issue #1195 

```
let tasks = {
     one: function(callback) {
         setTimeout(function() {
             callback(null, 'one');
         }, 200);
     },
     two: function(callback) {
         callback('two');
     },
     three: function(callback) {
         setTimeout(function() {
             callback(null, 'three');
         }, 100);
     }
 };

 async.parallel(async.reflectAll(tasks),
 // optional callback
 function(err, results) {
     // values
     // results.one.value = 'one'
     // results.two.error = 'two'
     // results.three.value = 'three'
 });
```

If you have any feedback just let me know, thanks.

### CODE REVIEW

**Comment by aearly** [comment]
Cool, great work!

File: lib/reflectAll.js [hunk 1]
```diff
@@ -38,7 +39,43 @@ import reflect from './reflect';
...
  */
 export default function reflectAll(tasks) {
-    return tasks.map(reflect);
+    var results;
+    if (isArray(tasks)) {
+        results = tasks.map(reflect);
+    } else {
+        var keys = Object.keys(tasks);
+        results = {};
+        keys.forEach(function(key) {
```
Comment (megawac) [inline lib/reflectAll.js]: change to `mapValues`

File: lib/reflectAll.js [hunk 2]
```diff
@@ -38,7 +39,43 @@ import reflect from './reflect';
...
+ *     // results.one.value = 'one'
+ *     // results.two.error = 'two'
+ *     // results.three.value = 'three'
+ * });
  */
 export default function reflectAll(tasks) {
-    return tasks.map(reflect);
+    var results;
+    if (isArray(tasks)) {
+        results = tasks.map(reflect);
```
Comment (megawac) [inline lib/reflectAll.js]: use `lodash/arrayMap`

File: lib/reflectAll.js [hunk 3]
```diff
@@ -38,7 +39,43 @@ import reflect from './reflect';
...
+    } else {
+        var keys = Object.keys(tasks);
+        results = {};
+        keys.forEach(function(key) {
+            results[key] = reflect.call(this, tasks[key]);
```
Comment (MaerF0x0) [inline lib/reflectAll.js]: Pretty sure this line doesnt work with async.auto  when a function has a dependency

**Pull request merged**